### ISSUE TITLE
Evaluate reducing required java version

### ISSUE THREAD

**Issue by phughk** [surrealdb/surrealdb.java#58]
We require users to use Java 17 - the latest LTS-supported Java language version and runtime.
However, internally in the driver, we probably aren't using any new functionality beyond Java 8. We can lower the required version of Java to support more projects.

[Java version support table](https://www.oracle.com/java/technologies/java-se-support-roadmap.html#:~:text=For%20product%20releases%20after%20Java,Java%2021%20in%20September%202023.) indicates that Java 8 can be used until 2030.

This issue is used both as a point of discussion and also as a ticket to undertake changing the required Java version.

**Comment by phughk** [comment]
Adding to this ticket - we probably want to keep 15 as the dev SDK and 8 as the binary SDK. Because we are using multi-line strings.

**Comment by Divine1** [comment]
@phughk 

Spring boot which is preferred by most folks supports [Jdk 17 and greater version](https://docs.spring.io/spring-boot/docs/current/reference/html/getting-started.html#getting-started.system-requirements) from 3.0.0 version.

is it ok to assume folks who wants to use latest technologies (For eg. surrealdb) would start with jdk 17 and not previous versions?

**Comment by phughk** [comment]
It's a difficult one to answer. Java 8 has a really long support timeline. We always have the option to support whichever versions we want. But if we want to be able to onboard legacy codebases, then supporting older versions is beneficial.

Not all projects will be new - some may be splinter projects from a shared codebase that are struggling to upgrade java versions. So the earlier the supported version the better. But obviously this has caveats, such as not being able to use the latest functionality. The latest APIs for native code invocation are particularly of interest, for example.

Basically this is still to be decided, but it is good to know about Spring 3 supported version - thank you for sharing that!

**Comment by emmanuel-keller** [comment]
The SurrealDB driver v0.2.0 supports every version of JAVA since 8.0
